source/linked_list.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/linked_list.h>
+#include <string.h>
+
+void aws_linked_list_init(struct aws_linked_list *list) {
+    list->head = 0;
+    list->length = 0;
+    list->tail = 0;
+}
+
+void aws_linked_list_clean_up(struct aws_linked_list *list) {
```

Comment:
Because linked lists are intrusive and can't be truly deallocated, this method isn't really meaningful.

Reply:
maybe we should just keep clear() and remove clean_up()?

Reply:
There is no clear() currently.

Reply:
There was and we removed it because of redundancy